ui/app/routes/vault/cluster/clients/counts/sync.ts

```diff
@@ -4,5 +4,57 @@
  */
 
 import Route from '@ember/routing/route';
+import { service } from '@ember/service';
+import { DEBUG } from '@glimmer/env';
 
-export default class ClientsCountsSyncRoute extends Route {}
+import type { ClientsCountsRouteModel } from '../counts';
+import type StoreService from 'vault/services/store';
+import ClientsActivityModel from 'vault/vault/models/clients/activity';
+
+interface ActivationFlagsResponse {
+  data: {
+    activated: Array<string>;
+    unactivated: Array<string>;
+  };
+}
+
+export default class ClientsCountsSyncRoute extends Route {
+  @service declare readonly store: StoreService;
+
+  async getActivatedFeatures() {
+    try {
+      const resp: ActivationFlagsResponse = await this.store
+        .adapterFor('application')
+        .ajax('/v1/sys/activation-flags', 'GET', { unauthenticated: true, namespace: null });
+      return resp.data?.activated;
+    } catch (error) {
+      if (DEBUG) console.error(error); // eslint-disable-line no-console
+      return [];
+    }
+  }
+
+  async secretsSyncActivated(activity: ClientsActivityModel | undefined) {
+    // if there are secrets, the feature is activated
+    if (activity && activity.total.secret_syncs > 0) return true;
```

Comment:
I _believe_ that `secret_syncs` should always exist on the response, but I'm not sure what the optin/enabled implications are on the total object here. Would be a good follow up question I can ask in a foundations sync, but for now we should probably add optional chaining here just in case `activity.total?.secret_syncs`